### DISCUSSION TITLE
Unable to retrieve all messages & attachments?

### DISCUSSION THREAD

**Discussion by tomwarham** [ReagentX/imessage-exporter#171] [Q&A] [answered]
Appologies if this isn't too technical. 

I can't seem to get message history & attachment export to go back much longer than a couple of months in the export. 

I've tried setting a date in the commend line such as 2018-01-01 as the start date with no luck. 

Its exporting roughly 800mb of attachments wit the -c efficient flag doing no attachment conversion, but the iPhone is stating there are 12gb worth of messages & files on the device. 

I've ensured local backup from the device is set to "back up all the data on your iPhone to this mac" but this also doesn't seem to help. It is stating the device backup is located in `~/Library/Application Support/MobileSync/Backup/<backup id>` which doesn't look to contain any .db files for messages.

Whereas the chat.db file which imessage-exporter seems to use is `~/Library/Messages/chat.db` which even more confusingly the attachments folder here is 1.65gb so something doesn't seem to be adding up. 

Any help is apreciated, and TL:DR; it feels like a case of pebdak on my part not being able to get the full chat.db file off my phone somehow. 

TIA

**Comment by ReagentX** [comment, marked as answer]
> Whereas the chat.db file which imessage-exporter seems to use is `~/Library/Messages/chat.db`

The software defaults to using the local macOS iMessage database. If you want to read from your iOS backup, specify the path to it in the command with `-p`. Per the docs:

```
-p, --db-path <path/to/source>
        Specify a custom path for the iMessage database location
        For macOS, specify a path to a `chat.db` file
        For iOS, specify a path to the root of an unencrypted backup directory
        If omitted, the default directory is ~/Library/Messages/chat.db
``` 

For example, a command like the following would export your iOS backup:

```zsh
imessage-exporter -f html -p ~/Library/Application Support/MobileSync/Backup/<backup id> -a iOS -c efficient
```

For more info, see some example commands [here](https://github.com/ReagentX/imessage-exporter/blob/develop/imessage-exporter/README.md#examples).

**Reply by michenator19** [reply]
Thank you, this was really helpful! I'm so close to figuring this out! I've moved my iPhone backup around a couple times but  wherever I save it I continue to get this error message: 

`Invalid configuration: Database not found at /Users/michele/Desktop/iphone_backup_latest/00008101-001A15682644001E/3d/3d0d7e5fb2ce288813306e4d4636395e047a3d28`

I feel like I've hit a dead end here, any suggestions? My backup is not encrypted.

**Reply by ReagentX** [reply]
What is the path to where your iOS backup is stored?

**Reply by michenator19** [reply]
It's in that 00008101 etc. folder inside the iphone_backup_latest folder on the Desktop. I moved it out of Library/Application Support/MobileSync/Backup because the commands in Terminal seemed (to me at least!) to get tripped up with the fact that Application Support was two words. Anyway, everything I'm looking up says that 3d0d7e5 etc. file should have a database _inside_ it, despite the fact that it's not a .db file. But my few attempts here keep coming up with "database not found."

**Reply by ReagentX** [reply]
You should point `imessage-exporter` at the root of the backup file, not inside of it.

**Reply by michenator19** [reply]
I believe I did that?? Here is the command I put in: 

`imessage-exporter -f html -p ~/Desktop/iphone_backup_latest/00008101-001A15682644001E -a iOS -c compatible -o ~/Desktop/iMessages1`

Here is the response I get back: 

`Invalid configuration: Database not found at /Users/michele/Desktop/iphone_backup_latest/00008101-001A15682644001E/3d/3d0d7e5fb2ce288813306e4d4636395e047a3d28`

Thank you for helping!!

**Reply by ReagentX** [reply]
That looks right to me, can you create a new discussion post with some information about your system and the iOS backup (software versions, iOS version etc.)?
